### PR TITLE
[Swift] getTypeFromMangledSymbolname() => getTypeForMangling()

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes rdar://41115104.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
https://github.com/apple/swift/pull/22241
@swift-ci Please test

File: source/Symbol/SwiftASTContext.cpp [hunk 1]
```diff
@@ -3947,12 +3956,12 @@ SwiftASTContext::GetTypeFromMangledTypename(const char *mangled_typename,
...
                 this, mangled_typename);
 
-  std::string swift_error;
-  found_type = swift::ide::getTypeFromMangledSymbolname(
-                   *ast_ctx, mangled_typename, swift_error)
+  found_type = swift::Demangle::getTypeForMangling(*ast_ctx, mangled_typename)
                    .getPointer();
 
   if (found_type) {
+    found_type = convertSILFunctionTypesToASTFunctionTypes(found_type)
```
Comment (dcci) [inline source/Symbol/SwiftASTContext.cpp:3973]: Can you add a comment here?

**Review by dcci — APPROVED** [review]
LGTM

File: source/Symbol/SwiftASTContext.cpp (hunk 1 above)
Comment (slavapestov) [reply source/Symbol/SwiftASTContext.cpp:3973]: Good idea. I put the comment on the function itself

**Comment by slavapestov** [comment]
apple/swift#22241
@swift-ci Please test

**Comment by dcci** [comment]
No other comments from me. Thanks for working on this, Slava.

**Comment by slavapestov** [comment]
@dcci Pleasure is all mine :)

**Pull request merged**